internal/kafka/internal/handlers/validation.go

```diff
@@ -51,46 +51,109 @@ func ValidateKafkaClusterNameIsUnique(name *string, kafkaService services.KafkaS
 	}
 }
 
-// ValidateCloudProvider returns a validator that sets default cloud provider details if needed and validates provided
-// provider and region
-func ValidateCloudProvider(kafkaService *services.KafkaService, kafkaRequest *dbapi.KafkaRequest, providerConfig *config.ProviderConfig, action string) handlers.Validate {
-	return func() *errors.ServiceError {
-		// Set Cloud Provider default if not received in the request
-		supportedProviders := providerConfig.ProvidersConfig.SupportedProviders
-		if kafkaRequest.CloudProvider == "" {
-			defaultProvider, _ := supportedProviders.GetDefault()
-			kafkaRequest.CloudProvider = defaultProvider.Name
-		}
+func getCloudProviderAndRegion(
+	ctx context.Context,
+	kafkaService *services.KafkaService,
+	kafkaRequest *public.KafkaRequestPayload,
+	providerConfig *config.ProviderConfig) (string, string, *errors.ServiceError) {
 
-		// Validation for Cloud Provider
-		provider, providerSupported := supportedProviders.GetByName(kafkaRequest.CloudProvider)
-		if !providerSupported {
-			return errors.ProviderNotSupported("provider %s is not supported, supported providers are: %s", kafkaRequest.CloudProvider, supportedProviders)
-		}
+	// Set Cloud Provider default if not received in the request
+	supportedProviders := providerConfig.ProvidersConfig.SupportedProviders
 
-		// Set Cloud Region default if not received in the request
-		if kafkaRequest.Region == "" {
-			defaultRegion, _ := provider.GetDefaultRegion()
-			kafkaRequest.Region = defaultRegion.Name
-		}
+	defaultProvider, _ := supportedProviders.GetDefault()
```

Comment:
I guess the reaso is that if a default provider is not found, an empty provider instance is returned, together with the error.
Since the empty provider will have an empty name, the next `supportedProvider.GetByName` will fail.